pkg/cloudprovider/aws/amifamily/resolver.go

```diff
@@ -96,13 +100,9 @@ func (r Resolver) Resolve(ctx context.Context, provider *v1alpha1.AWS, nodeReque
 		return nil, err
 	}
 	amiFamily := GetAMIFamily(provider.AMIFamily, options)
-	amiIDs := map[string][]cloudprovider.InstanceType{}
-	for _, instanceType := range nodeRequest.InstanceTypeOptions {
-		amiID, err := r.amiProvider.Get(ctx, instanceType, amiFamily.SSMAlias(options.KubernetesVersion, instanceType))
-		if err != nil {
-			return nil, err
-		}
-		amiIDs[amiID] = append(amiIDs[amiID], instanceType)
+	amiIDs, err := r.amiProvider.Get(ctx, provider, nodeRequest, options, amiFamily)
```

Comment:
optional: you could inline `amiFamily` abovem without the performance issue that existed prior, now that you're no longer calling it in a loop.

Reply:
amiFamily gets used a few lines below too so can't inline it here I think